spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/datastore/GcpDatastoreAutoConfiguration.java

```diff
@@ -152,8 +162,16 @@ public DatastoreEntityConverter datastoreEntityConverter(DatastoreMappingContext
 
 	@Bean
 	@ConditionalOnMissingBean
-	public DatastoreTemplate datastoreTemplate(Datastore datastore, DatastoreMappingContext datastoreMappingContext,
+	public DatastoreTemplate datastoreTemplate(Supplier<? extends DatastoreReaderWriter> datastore,
+			DatastoreMappingContext datastoreMappingContext,
 			DatastoreEntityConverter datastoreEntityConverter, ObjectToKeyFactory objectToKeyFactory) {
 		return new DatastoreTemplate(datastore, datastoreEntityConverter, datastoreMappingContext, objectToKeyFactory);
 	}
+
+	/**
+	 * This interface is the return type for the bean that provides namespaces.
+	 */
+	public interface NamespaceProvider extends Supplier<String> {
```

Comment:
Maybe rename to `DatastoreNamespaceProvider` for clarity?

Reply:
changed